test/integration/targets/known_hosts/tasks/main.yml

```
@@ -61,6 +61,27 @@
     - 'known_hosts.stdout_lines[5].startswith("# example.net")'
     - 'known_hosts.stdout_lines[-1].strip() == example_org_rsa_key.strip()'
 
+# https://github.com/ansible/ansible/issues/84952
+# test addition when ~/.ssh/ directory missing and path=default
+- name: move ~/.ssh/ directory away
+  command: mv ~/.ssh/ ~/.ssh_bak/
+
+- name: add a new host with default path
+  known_hosts:
+    name: example.org
+    key: "{{ example_org_rsa_key }}"
+    state: present
+  register: result
```

Even if this task fails, we could restore the ~/.ssh directory. The assert task below validates the task didn't fail.

```suggestion
  register: result
  ignore_errors: True
```